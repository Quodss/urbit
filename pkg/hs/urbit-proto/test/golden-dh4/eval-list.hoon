=>
|%
++  list
  ^-  $-($ $)
  |=  a=$
  $%  [%nil $~]
      [%con i=a t=(list a)]
  ==
--
=/  a  `(list @)`[%con 1 %con 2 %nil ~]
a
