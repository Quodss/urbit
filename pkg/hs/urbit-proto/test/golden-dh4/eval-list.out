PARSED

=>  |%
    ++  list
      ^-  $-($ $)
      |=  a=$
      $%  {$nil [$~]}
          {$con [i=a t=(list a)]}
      ==
    --
=/  a
  `(list @)`[%con 1 %con 2 %nil ~]
a

SOFT

=>  |%
    ++  list
      ^-  $-($ $)
      |=  a=$
      $:  ?(%7104878 %7237475)
          ^-  $
          ?:  ?=(%"nil" +2)
            $~
          ??  ?=(%"con" +2)
          [i=a t=(list a)]
      ==
    --
=+  a=`(list @)`[%con 1 %con 2 %nil ~]
a

CODE

=>  |%
    ++  list
      |=  *
      $:  ?(%7104878 %7237475)
          ?:  ?=(%"nil" +2)
            $~
          [i=+14 t=(list.+15 +14)]
      ==
    --
=+  ["con" 1 "con" 2 "nil" 0]
+2

TYPE

%-  #<  |=  *
        $:  ?(%7104878 %7237475)
            ?:  ?=(%"nil" +2)
              $~
            [i=+14 t=(list.+15 +14)]
        ==
    =>  0
    |%
    ++  list
      |=  *
      $:  ?(%7104878 %7237475)
          ?:  ?=(%"nil" +2)
            $~
          [i=+14 t=(list.+15 +14)]
      ==
    --
@

BASE

["con" 1 "con" 2 "nil" 0]